server/routes/search/searchController.ts

```diff
@@ -13,48 +21,126 @@ export default class SearchController {
+      searchResults: pagedResults.content,
+      form: req.session.prisonerSearchForm,
+      paginationParams,
+    })
   }
 
   async postSearch(req: Request, res: Response): Promise<void> {
     logger.debug('post search /')
     // TODO Validate form
+    req.session.searchParams ??= {}
 
-    const prisonerSearchForm = req.body
     req.session.prisonerSearchForm = { ...trimForm(req.body) }
-    logger.debug('searchform is', req.session.prisonerSearchForm)
-
-    if (prisonerSearchForm.searchType === 'name') {
-      const prisonersByNameForm = SearchController.toPrisonersByName(prisonerSearchForm)
-      const { content } = await this.historicalPrisonerService.findPrisonersByName(req.user.token, prisonersByNameForm)
-      res.render('pages/search', { searchResults: content, form: req.session.prisonerSearchForm })
-    } else if (prisonerSearchForm.searchType === 'identifier') {
-      const { content } = await this.historicalPrisonerService.findPrisonersByIdentifiers(
-        req.user.token,
-        prisonerSearchForm,
-      )
-      res.render('pages/search', { searchResults: content, form: req.session.prisonerSearchForm })
-    } else if (prisonerSearchForm.searchType === 'address') {
-      const { content } = await this.historicalPrisonerService.findPrisonersByAddressTerms(
-        req.user.token,
-        prisonerSearchForm,
-      )
-      res.render('pages/search', { searchResults: content, form: req.session.prisonerSearchForm })
+    logger.debug('search form is', req.session.prisonerSearchForm)
+
+    const pagedResults: PagedModelPrisonerSearchDto = await this.doSearch(req, res)
+
+    const paginationParams = this.getPaginationParams(req, pagedResults.page)
+    return res.render('pages/search', {
+      searchResults: pagedResults.content,
+      form: req.session.prisonerSearchForm,
+      paginationParams,
+    })
+  }
+
+  async doSearch(req: Request, res: Response): Promise<PagedModelPrisonerSearchDto> {
+    const { prisonerSearchForm } = req.session
+
+    switch (prisonerSearchForm.searchType) {
+      case 'name':
+        return this.historicalPrisonerService.findPrisonersByName(
+          req.user.token,
+          SearchController.toPrisonersByName(prisonerSearchForm),
+          req.session.searchParams.page,
+        )
+      case 'identifier':
+        return this.historicalPrisonerService.findPrisonersByIdentifiers(
+          req.user.token,
+          SearchController.toPrisonersByIdentifiers(prisonerSearchForm),
+          req.session.searchParams.page,
+        )
+      case 'address':
+        return this.historicalPrisonerService.findPrisonersByAddressTerms(
+          req.user.token,
+          SearchController.toPrisonersByAddress(prisonerSearchForm),
+          req.session.searchParams.page,
+        )
+      default:
+        throw new Error(`Unknown search type: ${prisonerSearchForm.searchType}`)
     }
   }
 
+  getPaginationParams(req: Request, page: PageMetaData): LegacyPagination {
```

Comment:
Not sure if this Pagination is the one we want - it looks like it is but odd that it's Legacy